src/lex.c

```diff
@@ -31,15 +33,44 @@ type_t getTokenType(char *tok){
 		else return oper_grtr;
 	}
 
-	if(isdigit(tok[0])) return oper_const;
+	if (tok[0] == '.' || isdigit(tok[0])) return oper_const;
 
 	// Namenstabelle prüfen
 	for (int i = 0; i < nameCount; i++) {
 		if (strcmp(tok, name_tab[i].name) == 0) return name_tab[i].type;
 	}
-	//printf("Type not found: %s\n", tok);
-	// ToDo: ggf. Namenstabelle hinzufügen
-//	return NULL;
+
+	//prüfen, ob zulässiger Variablen- oder Funktionsname, um es Namenstabelle hinzuzufügen
+	if ((tok[0] == '_' || isalpha(tok[0]) || tok[0] == '@') && (nameCount <= MAX_NAMES)) {
+			for(int i = 0; tok[i] != '\0'; i++){
+				//ToDo Verständnis: darf @ nur am Anfang stehen?
+				if(isalpha(tok[i]) || isdigit(tok[i]) || tok[i] == '_' || (i == 0 && tok[0] == '@')) {
+					continue;
+				}
+				else {
+					//kein zulässiger Variablen- oder Funktionsname -> Fehlermeldung & Abbruch
+					fprintf(stderr, "Unzulässiger Variablen- oder Funktionsname\n");
+					exit(EXIT_FAILURE);
+				}
+			}
+		type_t type = (tok[0] == '@') ? name_glob : name_any;
+		printf("type %d", type);
+		nameentry_t *name_entry = name_tab;
+		name_entry->type = type;
+		name_entry->name = tok;
+		if (type == name_glob) {
+			//das ist Quatsch.. muss der Wert sein, der nach dem "=" kommt -> eigentlich ja IMMER übernächstes Token, oder?
+			namedata_t d_name_entry;
+			d_name_entry.val = atof(tok);
+			name_entry->d = d_name_entry;
+		}
+		name_tab[nameCount] = *name_entry;
+		nameCount++;
+		return type;
```

Comment:
Bitte nochmal drüberschauen, aber ich denke so sollte das etwa passen:

```suggestion
		type_t type = (tok[0] == '@') ? name_glob : name_any;
		printf("type %d", type);
		nameentry_t *name_entry = name_tab[nameCount];
		name_entry->type = type;
		name_entry->name = tok; // ToDo: create Copy of string (tok is not persistent!)
		nameCount++;
		return type;
```

Reply:
Und wo wird d dann bei globalen Variabeln befüllt?

Reply:
> Und wo wird d dann bei globalen Variabeln befüllt?

Der Parser kümmert sich darum. Der Lexer sorgt nur dafür, das Token ausgeteilt werden und die Einträge in der Namenstabelle vorhanden sind.